layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging/0002-env_default.h-Add-support-for-OS_KERNEL_CMDLINE.patch

```diff
@@ -0,0 +1,77 @@
+@@ -23,8 +23,8 @@
+ 	"importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
+ 		"env import -t ${loadaddr} ${filesize}\0" \
+ 	"loadbootenv=fatload mmc ${mmcdev} ${loadaddr} ${bootenvfile}\0" \
+-	"loadimage=load ${devtype} ${bootpart} ${loadaddr} ${bootdir}/${bootfile}\0" \
+-	"loadfdt=load ${devtype} ${bootpart} ${fdtaddr} ${bootdir}/${fdtfile}\0" \
++	"loadimage=load ${devtype} ${resin_dev_index}:${resin_root_part} ${loadaddr} ${bootdir}/${bootfile}\0" \
++	"loadfdt=load ${devtype} ${resin_dev_index}:${resin_root_part} ${fdtaddr} ${bootdir}/${fdtfile}\0" \
+ 	"envboot=mmc dev ${mmcdev}; " \
+ 		"if mmc rescan; then " \
+ 			"echo SD/MMC found on device ${mmcdev};" \
+Index: git/include/environment/ti/mmc.env
+===================================================================
+--- git.orig/include/environment/ti/mmc.env
++++ git/include/environment/ti/mmc.env
+@@ -3,8 +3,7 @@ mmcrootfstype=ext4 rootwait
+ finduuid=part uuid ${boot} ${bootpart} uuid
+ args_mmc=run finduuid;setenv bootargs console=${console}
+ 	${optargs}
+-	root=PARTUUID=${uuid} rw
+-	rootfstype=${mmcrootfstype}
++	${resin_kernel_root}
+ loadbootscript=load mmc ${mmcdev} ${loadaddr} boot.scr
+ bootscript=echo Running bootscript from mmc${mmcdev} ...;
+ 	source ${loadaddr}
+@@ -12,9 +11,9 @@ bootenvfile=uEnv.txt
+ importbootenv=echo Importing environment from mmc${mmcdev} ...;
+ 	env import -t ${loadaddr} ${filesize}
+ loadbootenv=fatload mmc ${mmcdev} ${loadaddr} ${bootenvfile}
+-loadimage=load ${devtype} ${bootpart} ${loadaddr} ${bootdir}/${bootfile}
+-loadfdt=load ${devtype} ${bootpart} ${fdtaddr} ${bootdir}/dtb/${fdtfile}
+-get_fdt_mmc=load mmc ${bootpart} ${fdtaddr} ${bootdir}/dtb/${name_fdt}
++loadimage=load ${devtype} ${resin_dev_index}:${resin_root_part} ${loadaddr} ${bootdir}/${bootfile}
++loadfdt=load ${devtype} ${resin_dev_index}:${resin_root_part} ${fdtaddr} ${bootdir}/dtb/${fdtfile}
++get_fdt_mmc=load mmc ${resin_dev_index}:${resin_root_part} ${fdtaddr} ${bootdir}/dtb/${name_fdt}
+ envboot=mmc dev ${mmcdev};
+ 	if mmc rescan; then
+ 		echo SD/MMC found on device ${mmcdev};
+@@ -66,11 +65,11 @@ get_overlay_mmc=
+ 	fdt resize 0x100000;
+ 	for overlay in $name_overlays;
+ 	do;
+-	load mmc ${bootpart} ${dtboaddr} ${bootdir}/dtb/ti/${overlay} &&
++	load mmc ${resin_dev_index}:${resin_root_part} ${dtboaddr} ${bootdir}/dtb/ti/${overlay} &&
+ 	fdt apply ${dtboaddr};
+ 	done;
+-get_kern_mmc=load mmc ${bootpart} ${loadaddr}
++get_kern_mmc=load mmc ${resin_dev_index}:${resin_root_part} ${loadaddr}
+ 	${bootdir}/${name_kern}
+-get_fit_mmc=load mmc ${bootpart} ${addr_fit}
++get_fit_mmc=load mmc ${resin_dev_index}:${resin_root_part} ${addr_fit}
+ 	${bootdir}/${name_fit}
+ partitions=name=rootfs,start=0,size=-,uuid=${uuid_gpt_rootfs}
+Index: git/board/ti/j721e/j721e.env
+===================================================================
+--- git.orig/board/ti/j721e/j721e.env
++++ git/board/ti/j721e/j721e.env
+@@ -1,3 +1,4 @@
++#include <env_resin.h>
```

Comment:
Does https://github.com/balena-os/meta-balena/commit/281a2c8bd47bb77ccf1974ae05b9c42129094911 not work for this board?

Reply:
Inheriting resin-u-boot was not enough, indeed.

Reply:
So for this board it does not cascade to including include/env_default.h ?

Reply:
![Like this](https://github.com/balena-os/balena-beaglebone/assets/145996/f69660b5-24f1-4315-8d27-3d7502fff737)

Reply:
Probably, I have not dig any deeper.